src/main/java/com/ft/methodecontentcollectionmapper/mapping/EomFileToContentCollectionMapper.java

```diff
@@ -1,133 +1,181 @@
+			return resolveToBlogItem(attributesDocument, linkedObjectUUID, tid);
+		}
+
+		// this is a CPH referencing an internal content
+		try {
+			client.canResolveUUID(originalUUID, tid);
+			return new Item.Builder().withUuid(originalUUID).build();
+		} catch (IllegalArgumentException | UuidResolverException e) {
+			throw new ContentCollectionMapperException(String.format("Unable to map content package"), e);
+		}
+	}
+
+	private Item resolveToBlogItem(Document attributesDocument, String linkedObjectUUID, String tid)
+			throws XPathExpressionException, ParserConfigurationException, SAXException, IOException {
+		final XPath xpath = XPathFactory.newInstance().newXPath();
+		String listItemWiredIndexType = extractListItemWiredIndexType(xpath, attributesDocument);
+		if (!BLOG_CATEGORIES.contains(listItemWiredIndexType)) {
+			throw new UnsupportedTypeException(linkedObjectUUID, listItemWiredIndexType, "blog");
+		}
+
+		UUID fromStringUUID = UUID.fromString(linkedObjectUUID);
+		final String referenceId = extractRefField(xpath, attributesDocument, fromStringUUID);
+		final String guid = extractServiceId(xpath, attributesDocument, fromStringUUID);
+
+		return new Item.Builder().withUuid(blogUuidResolver.resolveUuid(guid, referenceId, tid)).build();
+	}
+
+	private DocumentBuilder getDocumentBuilder() throws ParserConfigurationException {
+		final DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+		documentBuilderFactory.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+
+		return documentBuilderFactory.newDocumentBuilder();
+	}
+
+	private boolean isContentPlaceholder(Document attributesDocument) throws XPathExpressionException {
+		final XPath xpath = XPathFactory.newInstance().newXPath();
+		final String listItemSrc = extractSource(xpath, attributesDocument);
+
+		return CONTENT_PLACEHOLDER_SRC.equals(listItemSrc);
+	}
+
+	private Document getLinkedObjectAttributes(final EomLinkedObject linkedObject)
+			throws ParserConfigurationException, SAXException, IOException {
+		final String itemAttributes = linkedObject.getAttributes();
+		if (Strings.isNullOrEmpty(itemAttributes)) {
+			return null;
+		}
+		return getDocumentBuilder().parse(new InputSource(new StringReader(itemAttributes)));
+	}
+
+	private String extractSource(XPath xPath, Document attributesDocument) throws XPathExpressionException {
+		return xPath.evaluate(XPATH_LIST_ITEM_SOURCE, attributesDocument);
+	}
+
+	private String extractOriginalUUID(XPath xPath, Document attributesDocument) throws XPathExpressionException {
+		return xPath.evaluate(XPATH_PLACEHOLDER_ORIGINAL_UUID, attributesDocument);
+	}
+
+	private String extractListItemWiredIndexType(XPath xPath, Document attributesDocument)
+			throws XPathExpressionException {
+		return xPath.evaluate(XPATH_LIST_ITEM_TYPE, attributesDocument);
+	}
+
+	private String extractServiceId(XPath xPath, Document attributesDocument, UUID uuid)
```

Comment:
`extractServiceId` and `extractRefField` methods could be refactored into one method that receives the XPATH as a parameter.